iOS_SDK/OneSignalSDK/Source/OSMessagingController.m

```diff
@@ -328,14 +469,19 @@ - (void)messageViewDidSelectAction:(OSInAppMessage *)message withAction:(OSInApp
     if (self.actionClickBlock)
         self.actionClickBlock(action);
   
+    let clickId = action.clickId;
+    
     // Make sure no click tracking is performed for IAM previews
     // If the IAM clickId exists within the cached clickedClickIds return early so the click is not tracked
+    // unless that click is from an IAM with redisplay
     // Handles body, button, or image clicks
-    if (message.isPreview || [self.clickedClickIds containsObject:action.clickId])
+    if (message.isPreview || [self isClickAvailable:message withClickId:clickId])
```

Comment:
This reads very confusing now. We want to return early on message `isPreview` and click is not available. The method name is confusing though, I understand the internals are actually checking the click is NOT available but we need to fix the name. As I suggested before:
We should use it like this in the if statement with the `isPreview` check`![self isClickAvailable:message withClickId:clickId]`

and internals:
```
/*
 * Checks for 2 important cases to determine if a click being available:
 * 1. Redisplay is enabled
 * 2. Click clickId should not be inside of the clickedClickIds set
 */
- (BOOL)isClickAvailable:(OSInAppMessage *)message withClickId:(NSString *)clickId {
    return [message.displayStats isRedisplayEnabled] && [message isClickAvailable:clickId]
}
```